### ISSUE TITLE
Remove the need for simapp in 08-wasm module

### ISSUE THREAD

**Issue by damiannolan** [cosmos/ibc-go#7771]
## Summary

It should be possible to refactor the tests in 08-wasm such that it does not rely on needing an entire simapp.

We only use a single test chain in 08-wasm and what we depend on mostly is the `MockVM` implementation. 
Investigate removal of the simapp in 08-wasm and adapting tests to not depend so heavily on it.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned
- [ ] Estimate provided

**Comment by duonghb53** [comment]
Hi @damiannolan 
Can I get this one?

**Comment by damiannolan** [comment]
Hey @duonghb53, I'm not longer maintainer on this repo. Maybe @AdityaSripal or @gjermundgaraba can chime in.

**Comment by duonghb53** [comment]
> Hey [@duonghb53](https://github.com/duonghb53), I'm not longer maintainer on this repo. Maybe [@AdityaSripal](https://github.com/AdityaSripal) or [@gjermundgaraba](https://github.com/gjermundgaraba) can chime in.

Thank @damiannolan , @gjermundgaraba can I this one?

**Comment by gjermundgaraba** [comment]
Feel free to give it a go @duonghb53! Let me know if you have any questions on implementation

**Comment by duonghb53** [comment]
@gjermundgaraba Yes. I'm going read more and ask later.

**Comment by gjermundgaraba** [comment]
From main

**Comment by duonghb53** [comment]
Hi @gjermundgaraba 
I recently reviewed the testing section in module 08-wasm. Do you have any documentation or a discussion thread about removing simapp? I’m unsure about the scope of this issue. Could you please provide more details? Thanks!

**Comment by gjermundgaraba** [comment]
No, but the issue gives a hint: the goal is to remove the need for the 08-wasm tests to have a full simapp. Our other tests manage without it, so the goal here would be to get 08-wasm's tests to the same point.
